### PR TITLE
Fix kid key rotation for OpenID Connect

### PR DESCRIPTION
**Pull request by davidhalter**
## Proposed changes

Fixes https://github.com/python-social-auth/social-core/issues/591. It fixes how kid's are used to fetch the correct keys (especially when keys are rotating).

## Types of changes

Please check the type of change your PR introduces:

- [ ] Release (new release request)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (PEP8, lint, formatting, renaming, etc)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes (build process, tests runner, etc)
- [x] Other (please describe): OpenID Connect Test Updates

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask. We're here to
help! This is simply a reminder of what we are going to look for before merging
your code._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: social_core/tests/backends/open_id_connect.py [hunk 1]
```diff
@@ -177,5 +193,41 @@ def test_invalid_issue_time(self):
...
+    def test_invalid_kid(self):
+        self.authtoken_raised('Token error: Signature verification failed', kid='doesnotexist')
+
+
+class ExampleOpenIdConnectAuth(OpenIdConnectAuth):
+    name = 'example123'
+    OIDC_ENDPOINT = 'https://example.com/oidc'
+
+
+class OpenIdConnectTest(OpenIdConnectTestMixin, OAuth2Test):
```
Comment (nijel) [inline social_core/tests/backends/open_id_connect.py:209]: Tests are not executed from this file, move it to `test_*.py`, for example  `social_core/tests/backends/test_open_id_connect.py`.

Comment (davidhalter) [reply social_core/tests/backends/open_id_connect.py:209]: Oh, good find! Thanks for bringing this up.

I guess it was confused that pytest actually executed the tests when I was specifying `pytest social_core/tests/backends/open_id_connect.py` directly.

Comment (davidhalter) [reply social_core/tests/backends/open_id_connect.py:209]: Fixed it.

Comment (nijel) [reply social_core/tests/backends/open_id_connect.py:209]: The mixin was probably intentionally in a non-matching filename - it does not make sense to execute it standalone. But it does no harm to have it in matching one as it does not inherit from UnitTest class...

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**